cli: add \x command to toggle records display format

erikgrinaker opened the pull request:
`psql` has a `\x` command to toggle extended output for results. CRDB
already supports this output format via `\set display_format=records`.
This commit adds `\x [on|off]` to toggle between `records` and `table`
display formats.

The `auto` option from `psql` to automatically enable extended output
depending on the result width is not supported, only `on` and `off`.

Resolves #56706.

Release note (cli change): A `\x [on|off]` command has been added to
toggle the `records` display format, following `psql` behavior.

Also adds support for `yes` and `no` as boolean options for slash commands as a separate commit, following `psql`.

erikgrinaker committed:
cli: add \x command to toggle records display format

`psql` has a `\x` command to toggle extended output for results. CRDB
already supports this output format via `\set display_format=records`.
This commit adds `\x [on|off]` to toggle between `records` and `table`
display formats.

The `auto` option from `psql` to automatically enable extended output
depending on the result width is not supported, only `on` and `off`.

Resolves #56706.

Release note (cli change): A `\x [on|off]` command has been added to
toggle the `records` display format, following `psql` behavior.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56829)
<!-- Reviewable:end -->

knz reviewed (commented):
Hi Eric, glad to see your contribution!

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56829#-:-MMPtFCjEaJLjh-mzu1_:b-rl5jxp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @erikgrinaker)

___
*[pkg/cli/sql.go, line 318 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56829#-MMPt7n3-iIIsCQjliM0:-MMPt7n3-iIIsCQjliM1:b-nqp3ad) ([raw file](https://github.com/cockroachdb/cockroach/blob/a71af523802717728ed500fe8459ffd1bf7d840a/pkg/cli/sql.go#L318)):*
> ```Go
> 			val = strings.ToLower(strings.TrimSpace(val))
> 			switch val {
> 			case "false", "0", "off":
> ```

then reuse it here
___
*[pkg/cli/sql.go, line 489 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56829#-MMPt0oK7GhvWtyerTCC:-MMPt0oL9S9v-Nf1HjQT:b-wrahhw) ([raw file](https://github.com/cockroachdb/cockroach/blob/a71af523802717728ed500fe8459ffd1bf7d840a/pkg/cli/sql.go#L489)):*
> ```Go
> 		err = opt.set(val)
> 	} else {
> 		switch val {
> ```

Can you extract this boolean parsing code into a reusable function
___
*[pkg/cli/sql.go, line 1181 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56829#-MMPtAUX6Etcer7BXN_0:-MMPtAUX6Etcer7BXN_1:b-jzlpa4) ([raw file](https://github.com/cockroachdb/cockroach/blob/a71af523802717728ed500fe8459ffd1bf7d840a/pkg/cli/sql.go#L1181)):*
> ```Go
> 		case 2:
> 			switch strings.TrimSpace(strings.ToLower(cmd[1])) {
> 			case "on", "1", "true":
> ```

and also here

thanks


<!-- Sent from Reviewable.io -->

erikgrinaker committed:
cli: add yes/no as valid boolean strings for slash commands

This commit adds a utility function `parseBool` for parsing boolean
strings in SQL shell slash commands, and uses it instead of ad-hoc parsing
in individual commands.

It also adds support for `yes` and `no` as valid boolean values,
following `psql` behavior.

Release note (cli change): The SQL shell now accepts `yes`/`no` as
boolean options for slash commands, following `psql` behavior.

erikgrinaker commented:
Sure @knz, done! Also added support for `yes` and `no` for boolean options, since `psql` already does this. Split it out as a separate commit.

knz reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thanks

> Reviewed 3 of 4 files at r2, 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56829#-:-MMVC2NSCxjn8lNobU0S:b-v6ynew)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

erikgrinaker commented:
Could you ask bors to merge this please?

knz commented:
bors r+
